### PR TITLE
MAT-2383 med req primary path

### PR DESCRIPTION
**Pull request by ghost**
1. Complex primary code paths are not supported in UI anymore. 
2. UI will show a message if the primary path code is not supported
3. Add MedicationRequest.medication, MedicationAministration.medication, MedicationDispense.medication attributes support

Pull requests into Bonnie require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] This PR is into the correct branch.
- [x] JIRA ticket for this PR:
- [ ] JIRA ticket links to this PR
- [ ] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [ ] If UI changes have been made, google WAVE plug-in has been executed to ensure no 508 issues were introduced.
- [ ] Tests are included and test edge cases
- [ ] Tests have been run locally and pass (remember to update Gemfile when applicable)
- [ ] Test fixtures updated and documented as necessary ( see [internal wiki](https://gitlab.mitre.org/bonnie/internal-documentation/wikis/testing#test-fixtures) )
- [ ] Code coverage has not gone down and all code touched or added is covered. 
     * In rare situations, this may not be possible or applicable to a PR. In those situations:
         1. Note why this could not be done or is not applicable here: 
         2. Add TODOs in the code noting that it requires a test
         3. Add a JIRA task to add the test and link it here: 
- [ ] Automated regression test(s) pass

If JIRA tests were used to supplement or replace automated tests:
- [ ] JIRA test links:
- [ ] Justification for using JIRA tests:
- [ ] JIRA tests have been added to sprint

**Reviewer 1:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [ ] JIRA tests have been run and pass
- [ ] You agree with the justification for use of JIRA tests or have provided input on why you disagree

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [ ] JIRA tests have been run and pass
- [ ] You agree with the justification for use of JIRA tests or have provided input on why you disagree

### CODE REVIEW

File: app/assets/javascripts/helpers/data_criteria_helpers.js.coffee [hunk 1]
```diff
@@ -757,10 +765,88 @@
...
+            fhirResource?.doNotPerform = primitiveBoolean
+        types: ['Boolean']
+      },
+      {
+        path: 'medication'
+        title: 'medication'
+        getValue: (fhirResource) =>
+          if !fhirResource?.medication?
+            return null
+          if cqm.models.CodeableConcept.isCodeableConcept(fhirResource?.medication)
```
Comment (ghost) [inline app/assets/javascripts/helpers/data_criteria_helpers.js.coffee:835]: add UT

File: app/assets/javascripts/helpers/data_criteria_helpers.js.coffee [hunk 2]
```diff
@@ -757,10 +765,88 @@
...
+        valueSets: () -> [USCoreMedicationCodesValueSet.JSON]
       }
     ]
-    MedicationDispense: []
     MedicationRequest: [
+      {
+        path: 'doNotPerform'
+        title: 'doNotPerform'
+        getValue: (fhirResource) => fhirResource?.doNotPerform?.value
+        setValue: (fhirResource, primitiveBoolean) =>
```
Comment (ghost) [inline app/assets/javascripts/helpers/data_criteria_helpers.js.coffee]: Comment, not ready

File: app/assets/javascripts/helpers/data_criteria_helpers.js.coffee [hunk 3]
```diff
@@ -757,10 +765,88 @@
...
+        valueSets: () -> [FhirValueSets.REASON_MEDICATION_NOT_GIVEN_VS]
+      }
+    ]
+    MedicationDispense: [
+      {
+        path: 'medication'
+        title: 'medication'
+        getValue: (fhirResource) =>
+          if !fhirResource?.medication?
+            return null
```
Comment (ghost) [inline app/assets/javascripts/helpers/data_criteria_helpers.js.coffee:798]: add UT

File: app/assets/javascripts/helpers/data_criteria_helpers.js.coffee [hunk 4]
```diff
@@ -748,6 +734,28 @@
...
+          if cqm.models.CodeableConcept.isCodeableConcept(fhirResource?.medication)
+            return fhirResource.medication.coding?[0]
+          else
+            fhirResource?.medication
```
Comment (ghost) [inline app/assets/javascripts/helpers/data_criteria_helpers.js.coffee:746]: add UT

File: app/assets/javascripts/templates/patient_builder/unsupported_codes_display.hbs [hunk 5]
```diff
@@ -0,0 +1,5 @@
+<div class="unsupported-primarypath-codes">
+<span>
+    {{#if primaryCodePath}}{{primaryCodePath}} is complex and not supported in this section. Please use the attribute section below{{else}}No primary path exists for this resource{{/if}}
```
Comment (adongare) [inline app/assets/javascripts/templates/patient_builder/unsupported_codes_display.hbs]: Should there be a word code?
  _No primary **code** path exists for this resource_

**Review by adongare — DISMISSED** [review]

File: app/assets/javascripts/templates/patient_builder/unsupported_codes_display.hbs (hunk 5 above)
Comment (ghost) [reply app/assets/javascripts/templates/patient_builder/unsupported_codes_display.hbs]: Good catch! I will update the text

Comment (ghost) [reply app/assets/javascripts/templates/patient_builder/unsupported_codes_display.hbs]: @adongare  - Corrected. Could you please re-approve?

**Review by adongare — APPROVED** [review]

**Review by jkotanchik-SB — APPROVED** [review]
